akka-javasdk-tests/src/test/java/akkajavasdk/components/agent/TestModelProvider.java

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright (C) 2021-2024 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package akkajavasdk.components.agent;
+
+import akka.japi.Pair;
+import akka.javasdk.agent.ModelProvider;
+import dev.langchain4j.data.message.AiMessage;
+import dev.langchain4j.data.message.UserMessage;
+import dev.langchain4j.model.chat.ChatModel;
+import dev.langchain4j.model.chat.request.ChatRequest;
+import dev.langchain4j.model.chat.response.ChatResponse;
+import dev.langchain4j.model.output.FinishReason;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.function.Predicate;
+
+public class TestModelProvider implements ModelProvider.Custom {
+
+  private List<Pair<Predicate<String>, String>> responsePredicates = new ArrayList<>();
+
+
+  @Override
+  public Object createChatModel() {
+    return new ChatModel() {
+      @Override
+      public ChatResponse chat(ChatRequest chatRequest) {
+
+        var userMessageTexts = chatRequest.messages().stream()
```

Comment:
The `chatRequest.messages` will include all messages, also those from the conversational memory.
It's probably the last (not first) one that should be the input to the predicate.

Reply:
ok, I'm using the last one now